src/rantanplan/core.py

```diff
@@ -310,17 +424,21 @@ def get_scansion(text, rhyme_analysis=False):
             seen_tokens.append(token)
     if len(seen_tokens) > 0:
         lines.append({"tokens": get_syllables(seen_tokens)})
+    for line in lines:
+        line.update(get_rhythmical_pattern(line, rhythm_format))
     if rhyme_analysis:
-        for rhyme in analyze_rhyme(lines):
-            for index, line in enumerate(lines):
-                line["structure"] = rhyme["name"]
-                line["rhyme"] = rhyme["rhyme"][index]
-                line["ending"] = rhyme["endings"][index]
-                line["ending_stress"] = rhyme["endings_stress"][index]
-                if line["ending_stress"] == 0:
-                    line["rhyme_type"] = ""
-                    line["rhyme_relaxation"] = None
-                else:
-                    line["rhyme_type"] = rhyme["rhyme_type"]
-                    line["rhyme_relaxation"] = rhyme["rhyme_relaxation"]
+        analyzed_lines = analyze_rhyme(lines)
+        if analyzed_lines is not None:
+            for rhyme in [analyzed_lines]:
```

Comment:
This is so confusing. Once the PR is done, we should carefully review the code to make it a bit clearer.

Reply:
Yes I agree

Reply:
Adding an issue (#55)